CIScripts/Common/Invoke-CommandWithFunctions.ps1

```diff
@@ -0,0 +1,24 @@
+. $PSScriptRoot/Aliases.ps1
+function Invoke-CommandWithFunctions {
+    Param(
+        [Parameter(Mandatory=$true)] [ScriptBlock] $ScriptBlock,
+        [Parameter(Mandatory=$true)] [PSSessionT] $Session,
+        [Parameter(Mandatory=$true)] [string[]] $FunctionNames,
+        [Switch] $CaptureOutput
+    )
+
+    $FunctionsInvoked = $FunctionNames `
+        | ForEach-Object { @{ Name = $_; Body = Get-Content function:$_ } }
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        $Using:FunctionsInvoked | ForEach-Object { Invoke-Expression "function $( $_.Name ) { $( $_.Body ) }" }
+    }
+
+    $Output = Invoke-Command -Session $Session -ScriptBlock $ScriptBlock
+
+    Invoke-Command -Session $Session -ScriptBlock {
```

Comment:
Please make sure that this cleanup is invoked always (even if Invoke-Command throws an exception).